server/src/main/java/org/candlepin/pinsetter/core/PinsetterKernel.java

```diff
@@ -170,6 +171,22 @@ else if (!isClustered()) {
             log.debug("Jobs implemented:" + jobImpls);
             Set<JobKey> jobKeys = scheduler.getJobKeys(jobGroupEquals(CRON_GROUP));
 
+            /*
+             * purge jobs that have been deleted from this version of Candlepin.
+             * This is necessary as we might not even have the Class definition
+             * at classpath, Hence any attempt at fetching the JobDetail by the
+             * Scheduler or JobStatus by the JobCurator will fail.
+             */
```

Comment:
@alikins this was an option that was considered before I got into implementation, but the approach that ended up being implemented seemed more cleaner in the long run, basically leaving zombie - noop- job definitions vs. just hard coded class names.